lints/lint_sub_ca_eku_valid_fields.go

```diff
@@ -0,0 +1,41 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type subCAEKUValidFields struct {
+	// Internal data here
+}
+
+func (l *subCAEKUValidFields) Initialize() error {
+	return nil
+}
+
+func (l *subCAEKUValidFields) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubCA(c) && util.IsExtInCert(c, util.EkuSynOid)
+}
+
+func (l *subCAEKUValidFields) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	for _, ekuValue := range c.ExtKeyUsage {
+		if ekuValue == x509.ExtKeyUsageServerAuth ||
+			ekuValue == x509.ExtKeyUsageClientAuth {
+			continue
+		} else {
+			return ResultStruct{Result: Error}, nil
+		}
+	}
+	return ResultStruct{Result: Pass}, nil
+}
+
+func init() {
+	RegisterLint(&Lint{
+		Name:          "e_sub_ca_eku_valid_fields",
+		Description:   "Subordinate CA extkeyUsage, either id-kp-serverAuth or id-kp-clientAuth or both values MUST be present.",
```

Comment:
EKU fields are optional, you're right. However, the CheckApplies function in this lint basically checks if the EKU extensions exists in the cert. If it does exist, then they must be one of these fields. Does that make sense?